bigtable/use-cases/fraudDetection/src/main/java/bigtable/fraud/utils/RowDetails.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package bigtable.fraud.utils;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import org.apache.beam.sdk.coders.AvroCoder;
+import org.apache.beam.sdk.coders.DefaultCoder;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.util.Bytes;
+
+@DefaultCoder(AvroCoder.class)
+
+/*
+ * Facilitates reading/writing from/to Cloud Bigtable. Classes that extend this
+ * class can easily construct their objects by reading a comma-separated line,
+ * or reading from a Cloud Bigtable instance.
+ * Used to create an object with the following format:
+ * rowKey, [TimestampMilliseconds=VALUE], column1, column2, ... etc
+ * The TimestampMilliseconds field is optional.
+ * In order to use this class, member variables needs to be the same name as
+ * the columns read from Cloud Bigtable, and they should be in the same
+ * sequence when read from Cloud Pubsub.
+ */
+public abstract class RowDetails {
+
+  /**
+   * The timestamp (millisecond) to use when writing to Cloud Bigtable.
+   */
+  private long timestampMillisecond;
+
+  /**
+   * Constructs a RowDetails Object.
+   */
+  public RowDetails() {
+    timestampMillisecond = Long.MAX_VALUE;
+  }
+
+  /**
+   * Constructs an object by reading a line from Pubsub. Supports writing a
+   * custom timestamp.
+   *
+   * @param line a comma-seperated line used to build a RowDetails object.
+   */
+  public RowDetails(final String line) {
+    timestampMillisecond = Long.MAX_VALUE;
+    List<String> values = new LinkedList<>(Arrays.asList(line.split(", ")));
+
+    // If a custom timestamp is to be used, it should be the second field
+    // and in this format: TimestampMilliseconds=TIME
+    if (values.size() >= 2 && values.get(1)
+        .contains("TimestampMilliseconds=")) {
```

Comment:
Is `startsWith` more appropriate here instead of `contains`?

Reply:
Makes sense